### PR TITLE
advisor: merge roofline and json

### PR DESCRIPTION
**Pull request by georgebisbas**
Json files are merged with rooflines. Drops duplicate code, enhances rooflines with callstacks and memory access patterns.

Main improvements:
- Add Memory Access Pattern collections
- Stability for more platforms, so far was mostly tested on Azure ones.
- Added workarounds for a few bugs that may easily lead to no data collected.

### CODE REVIEW

**Comment by FabioLuporini** [comment]
Hey @jack-lascala , would you mind taking a quick look at these changes?

File: benchmarks/user/advisor/run_advisor.py [hunk 1]
```diff
@@ -100,21 +100,24 @@ def run_with_advisor(path, output, name, exec_args):
     ]
     advisor_cmd = [
         'advixe-cl',
-        '-q',  # Silence advisor
+        #'-q',  # Silence advisor
```
Comment (georgebisbas) [inline benchmarks/user/advisor/run_advisor.py:103]: I would propose to get rid of silencing cuz it also silences bugs and failures!

File: benchmarks/user/advisor/roofline.py [hunk 2]
```diff
@@ -101,6 +110,13 @@ def roofline(name, project, scale, precision, mode, th):
     gflops_min = 2**0
     width = ai_max
 
+    # Declare the dictionary that will hold the JSON information
```
Comment (FabioLuporini) [inline benchmarks/user/advisor/roofline.py:113]: "Declare the dictionary" is obvious...

File: benchmarks/user/advisor/roofline.py [hunk 3]
```diff
@@ -101,6 +110,13 @@ def roofline(name, project, scale, precision, mode, th):
...
+    roofline_data = {}
+
+    # Declare the two types of rooflines dictionaries
```
Comment (FabioLuporini) [inline benchmarks/user/advisor/roofline.py:116]: so this is to generate the four lines L1/L2/L3/DRAM? and likewise for compute (I don't remember what they are, but you get the point)?

**Review by FabioLuporini — COMMENTED** [review]
so `advisor_to_json` dropped because it was 90% redundant stuff? do I get it right?

**Comment by georgebisbas** [comment]
> so `advisor_to_json` dropped because it was 90% redundant stuff? do I get it right?

Yes, look here: https://github.com/devitocodes/devito/pull/1463#pullrequestreview-496364136

File: benchmarks/user/advisor/roofline.py (hunk 3 above)
Comment (georgebisbas) [reply benchmarks/user/advisor/roofline.py:116]: Yes, for compute...sp_flops, dp_flops...etc

File: benchmarks/user/advisor/roofline.py (hunk 2 above)
Comment (georgebisbas) [reply benchmarks/user/advisor/roofline.py:113]: WIll drop then..leftover

**Pull request merged**